cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -645,7 +645,13 @@ private TwillController startTwillApplication(TwillRunnerService twillRunner) {
                                                                                 getSystemServiceInstances()));
 
         if (cConf.getBoolean(Constants.COLLECT_CONTAINER_LOGS)) {
-          preparer.addLogHandler(new PrinterLogHandler(new PrintWriter(System.out)));
+          preparer.addLogHandler(new LogHandler() {
+            @Override
+            public void onLog(LogEntry entry) {
+              ch.qos.logback.classic.Logger logger = (ch.qos.logback.classic.Logger) LOG;
```

Comment:
I think it is a good idea to make sure that LOG is instance of logback Logger. If not log a warning message, and add PrintHandler.